frontend/public/components/routes/RoutePage.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Formik, FormikHelpers } from 'formik';
+import { RouteComponentProps } from 'react-router-dom';
+import { k8sCreateResource, k8sUpdateResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { ErrorPage404 } from '@console/internal/components/error';
+import { StatusBox, history } from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { RouteModel, ServiceModel } from '@console/internal/models';
+import { K8sResourceKind, referenceForModel, RouteKind } from '@console/internal/module/k8s';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { safeJSToYAML, safeYAMLToJS } from '@console/shared/src/utils/yaml';
+import { baseTemplates } from '../../models/yaml-templates';
+import { RouteForm } from './RouteForm';
+import { useTranslation } from 'react-i18next';
+import { convertEditFormToRoute, convertRouteToEditForm, routeValidationSchema } from './utils';
+import { RouteFormProps } from './create-route';
+
+type RouteFormValues = {
+  editorType: EditorType;
+  formData: RouteFormProps;
+  yamlData: string;
+};
+
+const defaultRouteYAML = baseTemplates.get(referenceForModel(RouteModel)).get('default');
+
+export type RoutePageProps = RouteComponentProps<{ ns?: string; name?: string }>;
+
+export const RoutePage: React.FC<RoutePageProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const namespace = match.params.ns;
+  const name = match.params.name;
+  const heading = name ? t('public~Edit Route') : t('public~Create Route');
+  const submitLabel = name ? t('public~Save') : t('public~Create');
+
+  const [route, routeLoaded, routeLoadError] = useK8sWatchResource<RouteKind>({
+    kind: RouteModel.kind,
+    name,
+    namespace,
+  });
+  const [services, servicesLoaded, serviceLoadError] = useK8sWatchResource<K8sResourceKind[]>({
+    kind: ServiceModel.kind,
+    namespace,
+    isList: true,
+  });
+  const loaded = routeLoaded && servicesLoaded;
+
+  const initialValues = React.useMemo(
+    () => ({
+      editorType: EditorType.Form,
+      yamlData: name
+        ? safeJSToYAML(route, 'yamlData', {
+            skipInvalid: true,
+          })
+        : defaultRouteYAML,
+      formData: name ? convertRouteToEditForm(services, route) : {},
+    }),
+    [name, route, services],
+  );
+
+  const handleSubmit = async (values: RouteFormValues, helpers: FormikHelpers<RouteFormValues>) => {
+    const data =
+      values.editorType === EditorType.Form
+        ? convertEditFormToRoute(values.formData, name && route)
+        : safeYAMLToJS(values.yamlData);
+
+    try {
+      const method = name ? k8sUpdateResource : k8sCreateResource;
```

Comment:
It's better if we can use patch to update the resource to avoid conflicts, although that increases the complexity. Maybe a follow-on issue?